portfolio/src/main/webapp/script.js

```diff
@@ -85,6 +79,17 @@ function showSlidePicture(n) {
   return newIndex;
 };
 
+/**
+ * Return correct index for a circular array.
+ */
+function circularIndex(n, items) {
```

Comment:
Would (n + items.length) % items.length work?

Also is it possible that this library already exists somewhere, and you can just use it? In this case you don't need to add tests. :)

Reply:
To be more generic, it can be more like

while (n < 0) n += items.length;

return n % items.length.

Depending how negative your offset can be.

Reply:
> To be more generic, it can be more like
> 
> while (n < 0) n += items.length;
> 
> return n % items.length.
> 
> Depending how negative your offset can be.

That is probably true that there is likely not going to be more than a few iterations of that loop. If the size is very small it could be possible to cause it to loop many times, should I consider that case? If so then it may be faster to do two cases.

Reply:
Have implemented the loop change as it simplifies logic and improves overall readability. Also this edge case is not possible in the current usage of these functions.